dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/persister/EnrollmentPersister.java

```diff
@@ -142,4 +146,14 @@ protected TrackerType getType()
     {
         return TrackerType.ENROLLMENT;
     }
+
+    @Override
+    protected void persistOwnership( TrackerPreheat preheat, ProgramInstance entity )
```

Comment:
It seems `TrackerOwnershipManager#assignOwnership` is a pretty heavy method with get, save and update operations. Just checking, is this called for every enrollment during bulk import, and have we considered the performance impact of this change? I.e. are we doing individual get/save/update per object?

Reply:
Very good point Lars. As it stands, yes this will be called for every new enrollment during bulk import.  
This fix was necessary for KS instance that is about to go into production in less then a week. So the fix is in 2.36 already.
But on investigating, there are more gaps in ownership in the new tracker importer for which a jira issue is being created. I will be looking at including most things in preheat without using this #assignOwnership method, when I am working on that new issue.